alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/EventScanner.java

```diff
@@ -64,10 +66,12 @@ public void run() {
   private void pollEvents() {
     scheduler.scheduleWithFixedDelay(
         () -> {
+          abortTimeoutEvent();
           saveUncompensatedEventsToCommands();
           compensate();
           updateCompensatedCommands();
-          deleteDuplicateSagaEndedEvents();
+          deleteDuplicateEvents();
```

Comment:
Why do you have the duplicated events ?  The scheduler work should be done without any interruption, what if the delete work is finished  but update transaction status work is failed.

Reply:
Multiple alphas handling the same event will create the duplicates. The situation you mention will be  handled at the next execution of scheduler or another alpha since all status are stored in db.

Reply:
why not lock event row when scanning for timeout event to avoid duplicate aborted event?

Reply:
the lock will be invalid during the query of one alpha, what if another alpha visit the same row after the lock released?

Reply:
we can check if such a event already exists. as long as there's no concurrent processing, it has much less chance running into duplication